contracts/src/LightClient.sol

```diff
@@ -341,4 +381,78 @@ contract LightClient is Initializable, OwnableUpgradeable, UUPSUpgradeable {
             revert NoChangeRequired();
         }
     }
+
+    /// @notice check if more than threshold blocks passed since the last state update before
+    /// L1 Block Number
+    /// @param blockNumber The L1 block number
+    /// @param threshold The number of blocks updates to this contract is allowed to lag behind
+    function lagOverEscapeHatchThreshold(uint256 blockNumber, uint256 threshold)
+        public
+        view
+        returns (bool)
+    {
+        uint256 updatesCount = stateUpdateBlockNumbers.length;
+
+        // Handling Edge Cases
+        // Edgecase 1: The block is in the future or in the past before HotShot was live
+        if (blockNumber > block.number || updatesCount < 3) {
+            revert InsufficientSnapshotHistory();
+        }
+
+        uint256 prevBlock;
+        bool prevUpdateFound;
+
+        uint256 i = updatesCount - 1;
+        while (!prevUpdateFound) {
+            if (stateUpdateBlockNumbers[i] <= blockNumber) {
+                prevUpdateFound = true;
+                prevBlock = stateUpdateBlockNumbers[i];
+            }
+
+            // We don't consider the lag time for the first two updates
+            if (i < 2) {
+                break;
+            }
+            i--;
+        }
+
+        // If no snapshot is found, we don't have enough history stored to tell whether HotShot was
+        // down.
+        if (!prevUpdateFound) {
+            revert InsufficientSnapshotHistory();
+        }
+
+        return blockNumber - prevBlock > threshold;
+    }
+
+    /// @notice get the number of L1 block updates
+    function getStateUpdateBlockNumbersCount() public view returns (uint256) {
+        return stateUpdateBlockNumbers.length;
+    }
+
+    /// @notice get the HotShot commitment at the specified block height
+    /// @param hotShotBlockHeight hotShotBlockHeight
+    function getHotShotCommitment(uint256 hotShotBlockHeight)
```

Comment:
@ImJeremyHe now that we have this function, we no longer need to worry about the L1 proof height and I think we can remove from the header.